### DISCUSSION TITLE
XRDP on Ubuntu Budgie / Windows Manager Config Problem

### DISCUSSION THREAD

**Discussion by Obsidiannite21** [neutrinolabs/xrdp#3279] [General]
Hello,

I'm attempting to install & configure XRDP on a headless Ubuntu Budgie VM. I've done this before on Mint without an issue but I can't get it to work on this distro. Here is what I've done.

apt install xrdp & dependencies 
created a .xsession file in my home directory that contains "budgie-session"
chmod +x .xsession

That didn't work so then I modified startwm.sh

test -x /etc/X11/Xsession && exec /etc/X11/Xsession /usr/bin/budgie-desktop

All of the above allows me to connect, login with my credentials but then the connection terminates. The error in the log is:

Windows manager exited quickly (1 sec). This could indicate a window manager config problem.

I tried adding budgie-desktop to the end of the startwm.sh file. 
I've logged out each time I've made a change because I've read that I can't be logged in & connect via RDP at the same time.
I've restarted the service after each change.
My next test would be to use a different desktop environment but that sorta defeats the purpose since my goal is to try out the budgie desktop.

Can anyone help? Am I missing something?

**Comment by matt335672** [comment]
Hi @Obsidiannite21 

There are a number of things that could be going on here. Here are some initial pointers.
1) Set up a separate xrdp user so you don't have to keep logging out and in, or even restarting the service.
2) Check the contents of `~/.xsession-errors`.
3) Try a dry-run of startwm.sh by calling it with `DISPLAY= sh -x /etc/xrdp/startwm.sh` to check it's doing what you think it is.

**Comment by Obsidiannite21** [comment]
@matt335672 Thank you for the response. I took the path of least resistance by downloading the LTS version. It works perfectly on that build so it has something to do with 24.1
